tuna/rocmlir/rocmlir_tables.py

```diff
@@ -188,22 +189,83 @@ def __repr__(self) -> str:
 
   def config_string(self):
     """Return config as a flag/value string suitable for tuningRunner.py."""
-    string = "conv "  # +++pf:  of course generalise for gemm
-    for field, value in self.to_dict().items():
-      flag = self.options[field]
-      if flag:
+    string = "conv"
+    if self.data_type == 'f16':
+      string += 'fp16'
+    elif self.data_type == 'i8':
+      string += 'int8'
+    elif self.data_type == 'bf16':
+      string += 'bfp16'
+    string += " "
+
+    # In options order for canonicalisation, kind of.
+    for field, flag in self.options.items():
+      value = getattr(self, field, None)
+      if value is not None and flag is not None:
         string += f"{flag} {value} "
-    string += "-m conv"
+    string += "-t 1"  # Dummy "enable timing" option.
     return string
 
+  def parse_line(self, line):
+    """Parse a command-line-style conv config into a ConvolutionConfig object."""
+
+    print(f"Parsing line {line}")
+
+    # '-t 1' means 'enable timing' which is confused with -t for type.
+    line = line.replace("-t 1", "")
+
+    # Convert the line ("conv -n 256 -c 1024 -H 14 ...") to dict of flag and value.
+    i = iter(line.split())
+    operation = next(i)
+    options = dict(zip(i, i))
+    #  print(f"options = {options}")
+
+    # Mapping of flag to field name.
+    # -F 1 -n 2 -c 1280 -H 32 -W 32 -k 640 -y 1 -x 1 -p 0 -q 0 -u 1 -v 1 -l 1 -j 1 -m conv -g 1 -t 1
+    fields = {
+        '-F': 'direction',
+        '-f': 'fil_layout',
+        '-I': 'in_layout',
+        '-O': 'out_layout',
+        '-n': 'batchsize',
+        '-c': 'in_channels',
+        '-H': 'in_h',
+        '-W': 'in_w',
+        '-k': 'out_channels',
+        '-y': 'fil_h',
+        '-x': 'fil_w',
+        '-p': 'pad_h',
+        '-q': 'pad_w',
+        '-u': 'conv_stride_h',
+        '-v': 'conv_stride_w',
+        '-l': 'dilation_h',
+        '-j': 'dilation_w',
+        '-g': 'group_size',
+        '-m': None
+    }
+    # kernel-repeats has no flag, but perfRunner.py uses 5.
+    # ConvConfiguration.fromCommandLine accepts but skips -m and -t.
+    #   -m is operation and -t is (I think) -time from MIOpenDriver.
+    # data_type is inferred from operation -- conv is f32, convfp16 is f16,
+    #   convbfp16 is bf16, convint8 is i8
+
+    self.data_type = 'f32'
+    if operation == 'convfp16':
+      self.data_type = 'f16'
+    elif operation == 'convint8':
+      self.data_type = 'i8'
+    elif operation == 'convbfp16':
+      self.data_type = 'bf16'
+    self.kernel_repeats = 1
+    for flag, value in options.items():
+      field = fields[flag]
+      if field:
+        setattr(self, field, value)
+
+
+class ResultsMixin():  # pylint: disable=too-many-instance-attributes
```

Comment:
The bulk of results manipulation is the same for both (all?) operations, so abstract it.